lib/src/realtime/src/realtime_channel_options.dart

```diff
@@ -12,6 +14,6 @@ class RealtimeChannelOptions extends RestChannelOptions {
   final List<ChannelMode>? modes;
 
   /// create channel options with a cipher, params and modes
-  RealtimeChannelOptions(Object cipher, {this.params, this.modes})
-      : super(cipher);
+  RealtimeChannelOptions({this.params, this.modes, CipherParams? cipher})
```

Comment:
Does this represent a breaking API change?

Reply:
Yes, it's a breaking change. 

- It moves the interface from an API which was bad (forcing the user to set a cipher if they wanted to set params or modes) to one where all these options are optional. 
- It was also quite unclear, what types can Cipher be?. I also change it from an `Object` which could be anything, to be more specific: CipherParams. 

**In practice, I would not worry about this breaking change.** This constructor isn't widely used right now by customers. On Android, modes are ignored, and params was used as CipherParams. In iOS, a crash occurs. It is a completely untested / unvalidated piece of functionality in the app. 

These bugs are fixed in https://github.com/ably/ably-flutter/pull/191/files
```
readValueFromJson(jsonMap, PlatformConstants.TxRealtimeChannelOptions.params, v -> o.cipherParams = (Map<String, String>) v);
```

Notice the params passed in from the dart side is being set as cipherParams... 🙈